nodejs/image.bzl

```diff
@@ -88,6 +86,42 @@ _dep_layer = rule(
     implementation = _dep_layer_impl,
 )
 
+def _npm_deps_runfiles(dep):
+    depsets = []
+    for pkg in dep[NodeRuntimeDepsInfo].pkgs:
+        if NpmPackageInfo in pkg:
+            depsets.append(pkg[NpmPackageInfo].sources)
+    return depset(transitive = depsets)
+
+def _npm_deps_layer_impl(ctx):
+    return lang_image.implementation(ctx, runfiles = _npm_deps_runfiles, emptyfiles = _emptyfiles)
+
+_npm_deps_layer = rule(
+    attrs = lang_image.attrs,
+    executable = True,
+    outputs = lang_image.outputs,
+    toolchains = lang_image.toolchains,
+    implementation = _npm_deps_layer_impl,
+)
+
+def _app_runfiles(dep):
+    depsets = []
+    for pkg in dep[NodeRuntimeDepsInfo].pkgs:
+        if NpmPackageInfo not in pkg:
+            depsets.append(pkg.files)
+    return depset(transitive = depsets)
+
+def _app_layer_impl(ctx):
+    return lang_image.implementation(ctx, runfiles = _runfiles, emptyfiles = _emptyfiles)
```

Comment:
where is the entrypoint now being set since you're not calling app_layer any longer?

Reply:
Not sure what you're referring to as the entrypoint, do you mean the `entry_point` that exists on a `nodejs_binary`?
Because that gets collected via the `DeafultInfo` of the `nodejs_binary` on `_app_layer_impl`.
Note the `./app/testdata/nodejs_image_binary.runfiles/io_bazel_rules_docker/testdata/nodejs_image_binary.sh` in the above test, that's the script called to start the process inside the container.

Reply:
Let me expand a little on my question.
Currently, if you run a nodehs_image target, the entrypoint to the container is set [here](https://github.com/bazelbuild/rules_docker/blob/master/lang/image.bzl#L208). When I print its value for` //container/nodejs:nodejs_image` the entrypoint to the container is set to `"/app/tests/container/nodejs/nodejs_image.binary"`
Now in your change, you are replacing the call to `app_layer` in nodejs/image/bzl for a call to `_app_layer`. the rule `_app_layer `is implemented by `_app_layer_impl` which in turn is calling `lang_image.implementation(ctx, runfiles = _runfiles, emptyfiles = _emptyfiles)`. This call to `lang_image.implementation` is not setting any value for the entrypoint of the container. So where is the value for the entrypoint for the nodejs_image now being set?

Reply:
Ah I think I understand, initally I though we'd need a app specific layer there. But turns out it's not needed. I've switched back to use the `app_layer` from `//lang:image.bzl`.
Which should fix up the entrypoint issues